### PR TITLE
TestOfflineVnode - Race condition in momhook file copy

### PR DESCRIPTION
**Pull request by latha-subramanian**
#### Bug/feature Description
Intermittent test failures in TestOfflineVode due to mom hook sync timing issues

#### Affected Platform(s)
* All Linux platforms

#### Cause / Analysis / Design
There is a race condition between the mom hook files copy and subsequent operations by the test.  The test does not check if the hook files were successfully copied to the mom, before expecting the fail actions. Delay in mom hook file copy causes tests to fail intermittently in some test bed machines.

#### Solution Description
Add checks in the test to confirm successful copy of the hook files to the mom, before expecting fail actions. 

#### Testing logs/output

[TestOfflinevnode_beforefix.txt](https://github.com/PBSPro/pbspro/files/2367189/TestOfflinevnode_beforefix.txt)

[TestOfflinevnode_afterfix.txt](https://github.com/PBSPro/pbspro/files/2367190/TestOfflinevnode_afterfix.txt)

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
   - [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
   - [ ] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [ ] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: test/tests/functional/pbs_offline_vnodes.py [hunk 1]
```diff
@@ -130,6 +130,12 @@ def verify_vnodes_state(self, expected_state):
     def tearDown(self):
         TestFunctional.tearDown(self)
 
+        # Delete the hooks created by the test
+        hooks_list = ["h1", "h2", "h3"]
+        for h in hooks_list:
+            if h in self.server.hooks:
+                self.server.manager(MGR_CMD_DELETE, HOOK, id=h, expect=True)
```
Comment (bhroam) [inline test/tests/functional/pbs_offline_vnodes.py]: revert_to_defaults() should delete all the site hooks.

File: test/tests/functional/pbs_offline_vnodes.py [hunk 2]
```diff
@@ -150,9 +156,18 @@ def test_single_mom_hook_failure_affects_vnode(self):
...
         self.create_multi_vnodes(1)
         self.create_mom_hook()
 
+        # Check if hook files were copied to mom
+        single_mom.log_match(
+            "h1.HK;copy hook-related file request received",
+            existence=True, starttime=start_time, interval=2)
+        single_mom.log_match(
+            "h1.PY;copy hook-related file request received",
+            existence=True, starttime=start_time, interval=2)
```
Comment (bhroam) [inline test/tests/functional/pbs_offline_vnodes.py]: Consider removing existence=True.  It is the default (and the other log_match() calls).

File: test/tests/functional/pbs_offline_vnodes.py (hunk 1 above)
Comment (latha-subramanian) [reply test/tests/functional/pbs_offline_vnodes.py]: Yes. revert_to_defaults() is deleting the hooks anyways. removing this code from the test.

File: test/tests/functional/pbs_offline_vnodes.py (hunk 2 above)
Comment (latha-subramanian) [reply test/tests/functional/pbs_offline_vnodes.py]: done

**Comment by latha-subramanian** [comment]
Latest Logs: [TestOfflineVnode_afterfix2.txt](https://github.com/PBSPro/pbspro/files/2372092/TestOfflineVnode_afterfix2.txt)

**Comment by latha-subramanian** [comment]
Thanks @Lisa, @bhroam. 
I have squashed and rebased. Ready for merge

**Comment by bhroam** [comment]
I merged in the PR.

**Comment by latha-subramanian** [comment]
Thanks!

**Pull request merged**